### DISCUSSION TITLE
Efficiently Handling Large Datasets with Locally Hosted LLM (Ollama) and PostgreSQL

### DISCUSSION THREAD

**Discussion by ajai1923** [community/community#147995] [Programming Help]
### Select Topic Area

General

### Body

I am working with a locally hosted LLM (Ollama with Llama 3.1) to process queries based on a large dataset stored in a PostgreSQL database (~1 million rows). I am fetching data in chunks from the database and passing it to the model for processing.

However, I encounter an error: "too many requests" when querying the model with this setup.

The model runs locally, and I need to optimize the data flow to prevent overloading it while generating responses efficiently.

How can I handle such large datasets with this setup? Are there best practices or techniques for optimizing the interaction between the database and the model to avoid errors like this? Any suggestions or insights would be appreciated.

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @ajai1923!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
